packages/runtime/container-runtime/src/test/blobManager.spec.ts

```diff
@@ -232,6 +232,12 @@ describe("BlobManager", () => {
             handlePs.push(handleP);
             await waitForBlob(blob);
         };
+
+        const onNoPendingBlobs = () => {
+            assert((runtime.blobManager as any).pendingBlobs.size === 0);
+        };
+
+        runtime.blobManager.on("noPendingBlobs", () => onNoPendingBlobs());
```

Comment:
IMO it's also worth adding some test coverage that validates how many times this method is invoked in a few different scenarios (e.g. single blob upload, concurrent upload of several blobs, serial upload of several blobs). Though you could spec the method reasonably such that over-invoking it was still allowed, I prefer to make more contractual guarantees.